### ISSUE TITLE
[Question] Is this line a bug?

### ISSUE THREAD

**Issue by kozer** [hkproj/pytorch-llama#7]
https://github.com/hkproj/pytorch-llama/blob/067f8a37fe36ac8b52dca9cc6f2a2e8d6aa372d6/inference.py#L90
I have a question about the above line. Shouldn't the logits go through a `softmax` here?
Is this line correct?
Thanks a lot!

**Comment by hkproj** [comment]
In this case we are only interested in the index corresponding to the maximum value. Applying the softmax won't alter this information, so we can also not apply the softmax. If I remember correctly, it was done in the same way also in the official code of Llama.

Umar
